### PR TITLE
Added validation rules to the system.xml reference

### PR DESCRIPTION
**Pull request by naitsirch**
## Purpose of this pull request

The current reference for the validation rules is very incomplete. Some imported rules that were missing have been added:

* alphanumeric
* integer
* letters-only
* no-whitespace
* time
* time12h

The rules can be found in /path/to/magento-root/app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js

## Affected DevDocs pages

* https://devdocs.magento.com/guides/v2.3/config-guide/prod/config-reference-systemxml.html
* https://devdocs.magento.com/guides/v2.4/config-guide/prod/config-reference-systemxml.html

## Links to Magento source code

- https://github.com/magento/magento2/blob/2.3/app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js

whatsnew
Added missing field validation rules to the  [system.xml reference](https://devdocs.magento.com/guides/v2.3/config-guide/prod/config-reference-systemxml.html) in the _Magento Configuration Guide_.

## Notes

If you like this PR, I'd be happy if you could add the "[hacktoberfest-accepted](https://hacktoberfest.digitalocean.com/details#details)" label to the PR.

### CODE REVIEW

**Review by BarnyShergold — APPROVED** [review]
Great addition

**Comment by BarnyShergold** [comment]
Might be worth adding a "What's New" on this item

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md [hunk 1]
```diff
@@ -358,6 +358,12 @@ The following validation rules are available:
...
 |---------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| `alphanumeric`                  | Allows letters, numbers, spaces or underscores only                                                                     |
+| `integer`                       | Enter a positive or negative non-decimal number                                                                         |
+| `letters-only`                  | Allows letters only. For example, fooBAR                                                                                |
+| `no-whitespace`                 | Disallows white spaces.                                                                                                 |
+| `time`                          | Allows a valid time, between 00:00 and 23:59. For example 15, 15:05 or 15:05:48                                         |
+| `time12h`                       | Allows a valid time in 12 hours format, between 00:00 am and 12:00 pm. For example 3 am, 11:30 pm, 02:15:00 pm          |
```
Comment (meker12) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:366]:
```suggestion
| `time12h`                       | Allows a valid time in 12-hour format, between 12:00 am and 11:59:59 pm. For example 3 am, 11:30 pm, 02:15:00 pm          |
```

File: src/guides/v2.3/config-guide/prod/config-reference-systemxml.md [hunk 2]
```diff
@@ -358,6 +358,12 @@ The following validation rules are available:
...
 |---------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| `alphanumeric`                  | Allows letters, numbers, spaces or underscores only                                                                     |
+| `integer`                       | Enter a positive or negative non-decimal number                                                                         |
+| `letters-only`                  | Allows letters only. For example, fooBAR                                                                                |
+| `no-whitespace`                 | Disallows white spaces.                                                                                                 |
+| `time`                          | Allows a valid time, between 00:00 and 23:59. For example 15, 15:05 or 15:05:48                                         |
+| `time12h`                       | Allows a valid time in 12 hours format, between 00:00 am and 12:00 pm. For example 3 am, 11:30 pm, 02:15:00 pm          |
```
Comment (meker12) [inline src/guides/v2.3/config-guide/prod/config-reference-systemxml.md:366]:
```suggestion
| `time12h`                       | Allows a valid time in 12-hour format, between 12:00 am and 11:59:59 pm. For example 3 am, 11:30 pm, 02:15:00 pm          |
```

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md [hunk 3]
```diff
@@ -358,6 +358,12 @@ The following validation rules are available:
 
 | Rule                            | Description                                                                                                             |
 |---------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| `alphanumeric`                  | Allows letters, numbers, spaces or underscores only                                                                     |
+| `integer`                       | Enter a positive or negative non-decimal number                                                                         |
+| `letters-only`                  | Allows letters only. For example, fooBAR                                                                                |
+| `no-whitespace`                 | Disallows white spaces.                                                                                                 |
+| `time`                          | Allows a valid time, between 00:00 and 23:59. For example 15, 15:05 or 15:05:48                                         |
```
Comment (meker12) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:365]:
```suggestion
| `time`                          | Allows a valid time in 24-hour format, between 00:00 and 23:59. For example 15, 15:05 or 15:05:48                                         |
```

File: src/guides/v2.3/config-guide/prod/config-reference-systemxml.md [hunk 4]
```diff
@@ -358,6 +358,12 @@ The following validation rules are available:
 
 | Rule                            | Description                                                                                                             |
 |---------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| `alphanumeric`                  | Allows letters, numbers, spaces or underscores only                                                                     |
+| `integer`                       | Enter a positive or negative non-decimal number                                                                         |
+| `letters-only`                  | Allows letters only. For example, fooBAR                                                                                |
+| `no-whitespace`                 | Disallows white spaces.                                                                                                 |
+| `time`                          | Allows a valid time, between 00:00 and 23:59. For example 15, 15:05 or 15:05:48                                         |
```
Comment (meker12) [inline src/guides/v2.3/config-guide/prod/config-reference-systemxml.md:365]:
```suggestion
| `time`                          | Allows a valid time in 24-hour format, between 00:00 and 23:59. For example 15, 15:05 or 15:05:48                                         |
```

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md [hunk 5]
```diff
@@ -358,6 +358,12 @@ The following validation rules are available:
 
 | Rule                            | Description                                                                                                             |
 |---------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| `alphanumeric`                  | Allows letters, numbers, spaces or underscores only                                                                     |
+| `integer`                       | Enter a positive or negative non-decimal number                                                                         |
+| `letters-only`                  | Allows letters only. For example, fooBAR                                                                                |
```
Comment (meker12) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:363]:
```suggestion
| `letters-only`                  | Allows letters only. For example, abcABC                                                                                |
```

File: src/guides/v2.3/config-guide/prod/config-reference-systemxml.md [hunk 6]
```diff
@@ -358,6 +358,12 @@ The following validation rules are available:
 
 | Rule                            | Description                                                                                                             |
 |---------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| `alphanumeric`                  | Allows letters, numbers, spaces or underscores only                                                                     |
+| `integer`                       | Enter a positive or negative non-decimal number                                                                         |
+| `letters-only`                  | Allows letters only. For example, fooBAR                                                                                |
```
Comment (meker12) [inline src/guides/v2.3/config-guide/prod/config-reference-systemxml.md:363]:
```suggestion
| `letters-only`                  | Allows letters only. For example, abcABC                                                                                |
```

**Review by meker12 — CHANGES_REQUESTED** [review]

**Comment by meker12** [comment]
@naitsirch  Other field descriptions in table have period at end of sentence. For consistency, please add periods to the field descriptions that you added.

**Review by naitsirch — COMMENTED** [review]
Looks good. Thanks for the addition.

File: src/guides/v2.3/config-guide/prod/config-reference-systemxml.md [hunk 7]
```diff
@@ -358,6 +358,12 @@ The following validation rules are available:
...
 |---------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| `alphanumeric`                  | Allows letters, numbers, spaces or underscores only                                                                     |
+| `integer`                       | Enter a positive or negative non-decimal number                                                                         |
+| `letters-only`                  | Allows letters only. For example, abcABC                                                                                |
+| `no-whitespace`                 | Disallows white spaces.                                                                                                 |
+| `time`                          | Allows a valid time in 24-hour format, between 00:00 and 23:59. For example 15, 15:05 or 15:05:48                                         |
+| `time12h`                       | Allows a valid time in 12-hour format, between 12:00 am and 11:59:59 pm. For example 3 am, 11:30 pm, 02:15:00 pm          |
```
Comment (meker12) [inline src/guides/v2.3/config-guide/prod/config-reference-systemxml.md:366]:
```suggestion
| `alphanumeric`                  | Allows letters, numbers, spaces or underscores only.                                                                     |
| `integer`                       | Enter a positive or negative non-decimal number.                                                                         |
| `letters-only`                  | Allows letters only. For example, `abcABC`.                                                                                |
| `no-whitespace`                 | Disallows white spaces.                                                                                                 |
| `time`                          | Allows a valid time in 24-hour format, between 00:00 and 23:59. For example `15`, `15:05` or `15:05:48`                                         |
| `time12h`                       | Allows a valid time in 12-hour format, between 12:00 am and 11:59:59 pm. For example `3 am`, `11:30 pm`, `02:15:00 pm`.          |
```

File: src/guides/v2.4/config-guide/prod/config-reference-systemxml.md [hunk 8]
```diff
@@ -358,6 +358,12 @@ The following validation rules are available:
...
 |---------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| `alphanumeric`                  | Allows letters, numbers, spaces or underscores only                                                                     |
+| `integer`                       | Enter a positive or negative non-decimal number                                                                         |
+| `letters-only`                  | Allows letters only. For example, abcABC                                                                                |
+| `no-whitespace`                 | Disallows white spaces.                                                                                                 |
+| `time`                          | Allows a valid time in 24-hour format, between 00:00 and 23:59. For example 15, 15:05 or 15:05:48                                         |
+| `time12h`                       | Allows a valid time in 12-hour format, between 12:00 am and 11:59:59 pm. For example 3 am, 11:30 pm, 02:15:00 pm          |
```
Comment (meker12) [inline src/guides/v2.4/config-guide/prod/config-reference-systemxml.md:366]:
```suggestion
| `alphanumeric`                  | Allows letters, numbers, spaces or underscores only.                                                                    |
| `integer`                       | Enter a positive or negative non-decimal number.                                                                         |
| `letters-only`                  | Allows letters only. For example, `abcABC`.                                                                                |
| `no-whitespace`                 | Disallows white spaces.                                                                                                 |
| `time`                          | Allows a valid time in 24-hour format, between 00:00 and 23:59. For example `15`, `15:05` or `15:05:48`.                                        |
| `time12h`                       | Allows a valid time in 12-hour format, between 12:00 am and 11:59:59 pm. For example `3 am`, `11:30 pm`, `02:15:00 pm`.          |
```

**Review by meker12 — CHANGES_REQUESTED** [review]

**Review by meker12 — APPROVED** [review]

**Comment by meker12** [comment]
running tests

**Pull request merged**